activerecord/lib/active_record/connection_adapters/postgresql/quoting.rb

```diff
@@ -43,7 +43,7 @@ def quote(value) # :nodoc:
 
         # Quotes strings for use in SQL input.
         def quote_string(s) # :nodoc:
-          @raw_connection.escape(s)
+          valid_raw_connection.escape(s)
```

Comment:
Didn't you comment above that `any_raw_connection` could be used for escaping and such?

Reply:
In general yes that's the theory, but despite not _needing_ to do any server-facing comms to produce the result, `pg` in particular does check the connection status and potentially raise `PG::BadConnection`. (Which isn't an indefensible behaviour... _technically_ you should use the connection you're about to query to determine escaping rules, and the server might impose different rules upon reconnect -- but personally it's not something I'd bother caring about if given the option.)